client/request.go

```diff
@@ -134,8 +134,8 @@ func (cli *Client) doRequest(ctx context.Context, req *http.Request) (serverResp
 
 		// Don't decorate context sentinel errors; users may be comparing to
 		// them directly.
-		switch err {
-		case context.Canceled, context.DeadlineExceeded:
+		type timeoutError interface{ Timeout() bool }
```

Comment:
This will no longer catch `context.Canceled`, do we need to? If not you can check [`net.Error`](https://godoc.org/net#Error).

Reply:
Oh! Yes, my mistake; updated to detect the `context.Canceled` separately